macros/light-sdk-macros/src/hasher.rs

```diff
@@ -17,28 +17,84 @@ pub(crate) fn hasher(input: ItemStruct) -> Result<TokenStream> {
         .named
         .iter()
         .filter(|field| {
-            !field.attrs.iter().any(|attr| {
-                if let Some(attr_ident) = attr.path().get_ident() {
-                    attr_ident == "skip"
-                } else {
-                    false
-                }
-            })
+            !field.attrs.iter().any(|attr| attr.path().is_ident("skip"))
         })
         .map(|field| {
             let field_name = &field.ident;
-            let truncate = field.attrs.iter().any(|attr| {
-                if let Some(attr_ident) = attr.path().get_ident() {
-                    attr_ident == "truncate"
+            let truncate = field.attrs.iter().any(|attr| attr.path().is_ident("truncate"));
+            let nested = field.attrs.iter().any(|attr| attr.path().is_ident("nested"));
+            let is_option = is_option_type(&field.ty);
+
+
+            Ok(if nested {
+                if is_option {
+                    quote! {
+                        match &self.#field_name {
+                            Some(value) => {
+                                let mut nested_hash = ::light_hasher::DataHasher::hash::<::light_hasher::Poseidon>(value)
+                                    .expect("Failed to hash nested field");
+                                nested_hash[0] = 1; // Mark as Some
+                                result.push(nested_hash.to_vec());
+                            }
+                            None => {
+                                result.push([0u8; 32].to_vec());
+                            }
+                        }
+                    }
                 } else {
-                    false
+                    quote! {
+                        let nested_hash = ::light_hasher::DataHasher::hash::<::light_hasher::Poseidon>(&self.#field_name)
+                            .expect("Failed to hash nested field");
+                        result.push(nested_hash.to_vec());
+                    }
                 }
-            });
-            if truncate {
+            } else if is_option && truncate {
+                quote! {
+                    match &self.#field_name {
+                        Some(value) => {
+                            let (mut bytes, _) = ::light_utils::hash_to_bn254_field_size_be(&value.as_bytes())
+                                .expect("Could not truncate to BN254 field size");
+                            bytes[0] = 1; // Mark as Some
+                            result.push(bytes.to_vec());
+                        }
+                        None => {
+                            result.push([0u8; 32].to_vec());
+                        }
+                    }
+                }
+            } else if is_option {
+                quote! {
+                    match &self.#field_name {
+                        Some(value) => {
+                            let bytes = value.to_le_bytes();
+                            // if bytes.len() == 32 {
```

Comment:
```suggestion
                            // if bytes.len() >= 32 {
```

And I think we should throw such an error :D